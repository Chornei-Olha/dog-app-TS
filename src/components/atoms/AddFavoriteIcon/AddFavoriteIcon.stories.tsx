import type { Meta, StoryObj } from '@storybook/react';
import { AddFavoriteIcon } from '.';

const meta = {
  title: 'Atoms/AddFavoriteIcon',
  component: AddFavoriteIcon,
  argTypes: {
    state: {
      options: ['hover', 'active', 'default'],
      control: 'radio'
    },
    size: {
      options: ['small', 'middle'],
      control: 'radio'
    }
  }
} satisfies Meta<typeof AddFavoriteIcon>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    state: 'default',
    size: 'middle'
  }
};
