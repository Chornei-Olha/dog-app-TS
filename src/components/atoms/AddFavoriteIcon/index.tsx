import AddFavoriteHoverIcon from '../../../assets/icons/AddFavoriteHoverIcon.svg';
import AddFavoriteActiveIcon from '../../../assets/icons/AddFavoriteActiveIcon.svg';
import AddFavoriteDefaultIcon from '../../../assets/icons/AddFavoriteDefaultIcon.svg';

type TypeSize = {
  small: number;
  middle: number;
};
const sizeIcon: TypeSize = {
  small: 20,
  middle: 30
};

type AddFavoriteIconProps = {
  state?: 'hover' | 'active' | 'default';
  size?: 'small' | 'middle';
};

export const AddFavoriteIcon = ({
  state = 'default',
  size = 'middle'
}: AddFavoriteIconProps) => {
  switch (state) {
    case 'hover': {
      return <img src={AddFavoriteHoverIcon} alt="" width={sizeIcon[size]} />;
    }
    case 'active': {
      return <img src={AddFavoriteActiveIcon} alt="" width={sizeIcon[size]} />;
    }
    default: {
      return <img src={AddFavoriteDefaultIcon} alt="" width={sizeIcon[size]} />;
    }
  }
};
