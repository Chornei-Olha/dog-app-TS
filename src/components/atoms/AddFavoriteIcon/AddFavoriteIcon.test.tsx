import { composeStories } from '@storybook/react';
import * as stories from './AddFavoriteIcon.stories';
import { testSnapshot } from '../../../test/utils/snapshot.utils';

describe('Icon', () => {
  const { Primary } = composeStories(stories);
  testSnapshot(<Primary />);
});
