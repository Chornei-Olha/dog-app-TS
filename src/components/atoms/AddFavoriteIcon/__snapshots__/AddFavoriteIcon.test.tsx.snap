// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFavoriteIcon should match snapshot Primary 1`] = `
<div>
  <img
    alt=""
    src="test-file-stub"
    width="30"
  />
</div>
`;
